### DISCUSSION TITLE
How to build libssh2 for windows using nmake

### DISCUSSION THREAD

**Discussion by Hrishikesh-Padale** [libssh2/libssh2#1129] [Q&A] [answered]
I am trying to build libssh2 for windows platform using nmake. Can anyone help with the steps and requirements?

Getting this error on running : nmake -f NMakefile

NMAKE : fatal error U1073: don't know how to make 'Release/'
Stop.

**Comment by vszakats** [comment]
`NMakefile` is minimally tested, and after an overhaul, so there is a good chance for issues. I haven't seen this one while testing though. Does it help if you create the folder manually?

Our CMake builds should support generating NMake Makefiles, if you want to give it a shot:
https://cmake.org/cmake/help/latest/manual/cmake-generators.7.html#id11

**Reply by Hrishikesh-Padale** [reply]
Tried creating the folder manually but got the same error. Thanks for the suggestion :)

**Reply by vszakats** [reply]
Thanks for testing it! In that case I recommend the CMake route.

**Comment by vszakats** [comment, marked as answer]
`NMakefile` was subsequently deleted in c515eed39870e8621c05e1593545abfc2ae5b1fe #1134.
Alongside pure GNU Make in 43485579609a833827736e8205fe29956900ed74 #1204.

It means `NMakefile` and `Makefile.mk` will be missing from our next
planned release.

We continue to focus on CMake and autotools:

CMake has seen a lot of updates in 2023. It's thouroughly tested in CI. Its
speed has also been improved by a large margin on Windows. It should be
able to generate NMake makefiles.
